### DISCUSSION TITLE
load-weights

### DISCUSSION THREAD

**Discussion by vkaragounis** [HIRO-group/marl-experiments#41] [Q&A] [answered]
Is it possible to provide further details for the parameter 'load-weights'. I would like to test an example after the training completion.
What is the weight file's name? Is it "episode_reward.csv"? Can you please provide an example? Thank you.

**Comment by Jmiceli17** [comment, marked as answer]
Hi! 
I apologize, the `load-weights` parameter is currently unused. I have a ticket to remove deprecated config parameters but just haven't gotten to it yet. 

After training, you should be seeing logs generated in a `nn/<experiment_name>` directory in the form of `.pt` files. These are the weights of your model. For example, if you're using [`ac-indepnedent-SUMO.py`](https://github.com/HIRO-group/marl-experiments/blob/main/ac-independent-SUMO.py) with parameter sharing, you will see a `.pt` file that was saved every number of [`max-cycles`](https://github.com/HIRO-group/marl-experiments/blob/main/experiments/sumo-3x3.config#L7).


Those files can be loaded using the `load_state_dict()` function in torch. An example of this can be seen in [`sumo_analysis.py`](https://github.com/HIRO-group/marl-experiments/blob/main/sumo_analysis.py#L203) 

Note though that it matters if you used parameter sharing or not during training. If you DID NOT, then the `.pt` files will have their agent index assigned to them (i.e. training round 39500 will produce logs of `39500-0.pt`, `39500-1.pt`, `39500-2.pt`, etc.). Example of loading these types of weights here [here](https://github.com/HIRO-group/marl-experiments/blob/main/sumo_analysis.py#L216)

**Comment by Jmiceli17** [comment]
I've opened the discussions page so I'm moving this thread there.

Feel free to provide more context with what you're trying to do and I can try to help out!

**Comment by vkaragounis** [comment]
Thanks for your help. I managed to run those commands below

python3 dqn-independent-ps.py   -c experiments/sumo-2x2-ac-independent-ps.config
python3 sumo_analysis.py        -c experiments/sumo-2x2-ac-independent-ps.config

with success and also watched the trained simulation working perfect. However when i tried some 
other methods (ex. dqn-independent-empathetic.py,  dqn-independent-greedy.py) i got an error:

File "/marl-experiments-main/dqn-independent-empathetic.py", line 372, in <module>
    rb[agent].put((obses[agent], actions[agent], rewards[agent], next_obses[agent], dones[agent]))
TypeError: tuple indices must be integers or slices, not str

Also i realized that  'global_obs' should always be TRUE because 'else' is missing. With 'True'
 i got the error:

 File "/marl-experiments-main/dqn-independent-empathetic.py", line 319, in <module>
    global_obs = np.hstack(list(obses.values()))
AttributeError: 'tuple' object has no attribute 'values'

Maybe the error occurs in more methods. I tested only 'dqn-independent-empathetic.py' and 'fdqn-independent-greedy.py'
Thanks again for your time.

**Reply by Jmiceli17** [reply]
Glad to hear you got those working! 

I'll have to look into `dqn-independent-empathetic.py` I'm not sure why you're seeing that error.

`global_obs` is set to False by default in all the experiment files (see [here](https://github.com/HIRO-group/marl-experiments/blob/main/experiments/sumo-2x2-ac-independent-ps.config#L8) for example)

I have not really tested global observations very much but I can take a look at it.
